src/evaluation/hierarchy_evaluator.py

```
@@ -0,0 +1,91 @@
+# src/evaluation/hierarchy_evaluator.py
+import json
+from pathlib import Path
+from src.inference import ResNetInference, MobileNetInference
+from src.evaluation.evaluator import ModelEvaluator, EvaluationConfig
+from src.optimal_class_mapping import map_prediction
+from dataset_preparation.utilities.datasets import DATASETS
+from metrics.metrics_api import compute_metrics
+
+class MappedModelWrapper:
+    def __init__(self, model):
+        self.model = model
+    def infer(self, image):
+        pred = self.model.infer(image)
+        mapped_id = map_prediction(pred[0]["class_id"])
+        return [{"class_id": mapped_id, "probability": pred[0]["probability"]}]
+
+def evaluate_all_hierarchies(y_true, y_pred, latencies):
+    results = {}
+    
+    # CropAndWeed
+    report = compute_metrics(y_true, y_pred, latencies)
+    results["CropAndWeed"] = report.model_dump()
+
+    
+    # Fine24, CropsOrWeed9, CropOrWeed2
+    for name in ["Fine24", "CropsOrWeed9", "CropOrWeed2"]:
+        dataset = DATASETS[name]
+        mapped_true = [dataset.get_mapped_id(label) if dataset.get_mapped_id(label) else 255 for label in y_true]
```

why do we have this `255` in here? The script should fail if mapping is wrong, here it's failing silently and we don't know is there're any issues